scc/delivery-with-carvel-app.hbs.md

```diff
@@ -0,0 +1,210 @@
+
+   ```yaml
+   ---
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: app-values
+   stringData:
+     values.yaml: |
+       ---
+       replicas: 2
+       hostname: app.mycompany.com
+   ```
+
+   > **Note:** You can skip this step if you want to use the default parameter values.
+
+2. For each Run cluster, create a `PackageInstall`. Reference the `Secret` you created in the previous step. Store the `PackageInstall` in your GitOps repository at `<package_name>/<run_cluster>/packageinstall.yaml`.
+
+   ```yaml
+   ---
+   apiVersion: packaging.carvel.dev/v1alpha1
+   kind: PackageInstall
+   metadata:
+     name: app
+   spec:
+     serviceAccountName: <run-cluster-ns-sa> # ServiceAccount on Run cluster with permissions to deploy Package, see "Set up Run Cluster Namespaces"
+     packageRef:
+       refName: app.default.tap # name of the Package
+       versionSelection:
+         constraints: 20230321004057.0.0 # version of the Package
+     values:
+     - secretRef:
+         name: app-values # Secret created in previous step
+   ```
+
+   > **Note:** If you want to continously deploy the latest version of your `Package`, you can set `versionSelection.constraints: >=0.0.0`
+   > **Note:** If you skipped creation of the `Secret`, omit the `values` key.
+
+3. Push the newly created `PackageInstalls` and `Secrets` to your GitOps repository.
+
+## Create App
+
+1. You need to give the Build cluster access to the Run clusters. On the Build cluster, for each Run cluster, create a `Secret` containing the Run cluster's kubeconfig:
+
+   ```console
+   kubectl create secret generic <run-cluster>-kubeconfig \
+       -n <build-cluster-ns> \
+       --from-file=value.yaml=<path-to-run-cluster-kubeconfig>
+   ```
+
+2. The Carvel `App` custom resource represents a collection of Kubernetes resources that kapp-controller can fetch and deploy to a cluster. The `App` will be pointed at the git repository branch where kapp-controller resources (e.g. `PackageRepository` and `Packages`) will be defined. By default, an `App` custom resource will sync the cluster with its fetch source every 30 seconds to prevent the cluster state from drifting from its source of truth, which is a git repository in this case. Create the following `App` on your Build cluster:
+
+   ```yaml
+   ---
+   apiVersion: kappctrl.k14s.io/v1alpha1
+   kind: App
+   metadata:
+     name: hello-app-app
+     namespace: <build-cluster-ns>
+   spec:
+     # specifies that app should be deployed authenticated via the
+     # given service account, found in this namespace
+     serviceAccountName: <SERVICE_ACCOUNT>
```

Comment:
Is this service account on the run cluster or  build cluster? Can we set both separately?

Reply:
Its specified in the note right after: Each App CR must specify either a service account (via `spec.serviceAccountName`) or, a `Secret` with kubeconfig contents for some Run cluster (via `spec.cluster.kubeconfigSecretRef.name`).